client/network/src/protocol/legacy_proto/tests.rs

```diff
@@ -275,9 +279,21 @@ fn basic_two_nodes_requests_in_parallel() {
 	// Generate random messages with or without a request id.
 	let mut to_send = {
 		let mut to_send = Vec::new();
+		let mut existing_ids = HashSet::new();
 		for _ in 0..200 { // Note: don't make that number too high or the CPU usage will explode.
-			let msg = (0..10).map(|_| rand::random::<u8>()).collect::<Vec<_>>();
-			to_send.push(Message::<Block>::ChainSpecific(msg));
+			let req_id = loop {
```

Comment:
Just to make sure I understand this correctly, you have witnessed a u64 collision repeatedly? Even with the birthday attack in mind, that is rather unlikely, no? Am I missing something?

https://en.wikipedia.org/wiki/Birthday_attack#Mathematics

Reply:
According to the diagram, this test would fail with a probability of about 1/10^15. So yeah, probably overkill. TBH, I thought that `RequestId` was `u32` (and I suppose the type may yet change), so it seems safer to protect against that. If you push for it I'll change it.